### PR TITLE
Make smart_markers multitenancy ready

### PR DESCRIPTION
**Pull request by NelsonVides**
Mainly makes smart_markers multitenancy ready. This also moves around some formatting like removing variable names in spec declarations or reordering some variables.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/70021) / small_tests / add88452e30666a2d56e20a0048ce0bf4b77d5a8
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3243/70021/small_tests.24.0.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3243/70021/small_tests.24.0.4/small/ct_run.mongooseim@localhost.2021-09-03_21.55.17/lib.mongooseim.logs/run.2021-09-03_21.55.18/suite.log.html)

File: src/mod_smart_markers.erl [hunk 1]
```diff
@@ -145,33 +152,56 @@ user_send_packet(Acc, _From, _To, _Packet) ->
...
-                           To :: jid:jid(),
-                           Packet :: exml:element(),
-                           TS :: integer()) -> [chat_marker()].
-extract_chat_markers(From, To, Packet, TS) ->
+-spec has_valid_markers(mongoose_acc:t(), jid:jid(), jid:jid(), exml:element()) ->
+    false | {true, mongooseim:host_type(), Markers :: [chat_marker()]}.
+has_valid_markers(Acc, From, To, Packet) ->
+    case is_valid_markers(Acc, From, To, Packet) of
+        false -> false;
+        {true, Markers} ->
```
Comment (chrzaszcz) [inline src/mod_smart_markers.erl:177]: Minor: I would get rid of `is_valid_markers`. Apart form the grammatical error (that was there before), it does not bring any extra value.
```suggestion
    case extract_chat_markers(Acc, From, To, Packet) of
        [] -> false;
        Markers ->
```

**Review by chrzaszcz — COMMENTED** [review]
Looks good. I proposed some minor refactoring.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/70158) / small_tests / 7aeb3b1b18482efa0add1aa411ea8e8d4a64116b
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3243/70158/small_tests.24.0.4) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3243/70158/small_tests.24.0.4/small/ct_run.mongooseim@localhost.2021-09-06_08.52.23/lib.mongooseim.logs/run.2021-09-06_08.52.24/suite.log.html)

**Review by chrzaszcz — APPROVED** [review]
Looks great!

**Pull request merged**